extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -220,17 +220,40 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 		return err
 	}
 
+	err = wait.Poll(1*time.Second, defaults.FiveMinuteTimeout, func() (bool, error) {
+		updatedClusterResp, err := client.Management.Cluster.ByID(clusterID)
+		if err != nil {
+			return false, nil
+		}
+
+		updatedClusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane = initialControlPlaneValue
+		updatedClusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker = initialWorkerValue
+
+		_, err = client.Management.Cluster.Update(clusterResp, updatedClusterResp)
+		if err != nil {
+			return false, nil
+		}
+
+		if updatedClusterResp.State == active {
+			return true, nil
```

Comment:
Having this all in a poll is confusing, we would update the cluster multiple time until it's active?

Reply:
Do you mean the poll itself or just this specific part is confusing?

Reply:
Oh, I think you know what you mean. Let me try to fix that.

Reply:
Resolved.